plugins/lookup/spec.py

```diff
@@ -191,31 +193,27 @@
 
 display = Display()
 
-
 class LookupModule(LookupBase):
-    def run(self, terms, variables, **kwargs):
+    def run(self, terms, **kwargs):
 
         res = []
         output = {}
-        try:
-            yang_file = terms[0]
-        except IndexError:
-            raise AnsibleError("the yang file must be specified")
-
-        yang_file = os.path.realpath(os.path.expanduser(yang_file))
+        defaults, annotations, schema_out_path = False, False, None
+        yang_file = kwargs.pop("yang_file", "")
         if not os.path.isfile(yang_file):
             raise AnsibleError("%s invalid file path" % yang_file)
 
         search_path = kwargs.pop("search_path", "")
-        annotations = kwargs.pop("annotations", "")
+        xml_schema = kwargs.pop("xml_schema", {})
```

Comment:
This represents the argpsec for lookup plugin. In case of `generate_spec` module, the `xml_schema` and `json_schema` are using suboptions to allow separate configuration option for JSON and XML but in case of lookup pluign suboptions support is not required to keep the lookup plugin simple. Hence this change is not required IMO